### ISSUE TITLE
Modal form loads on all pages

### ISSUE THREAD

**Issue by mindctrl** [easydigitaldownloads/edd-free-downloads#130]
The form loads on all pages, even if there are no downloads on the page. This can be seen on any pages on the EDD site, for example.

https://github.com/easydigitaldownloads/edd-free-downloads/blob/2.1.2/includes/template-actions.php#L22-L31

**Comment by mindctrl** [comment]
Would love to see this fixed asap. This adds extra cruft to pages, even on /checkout. And in cases where the CSS is disabled, or doesn't download due to an HTML timeout/error, it's visible.

**Comment by ghost1227** [comment]
It's actually intentional. We can predict pages like the actual download list page (if it's set in settings), and individual download pages, but we can't predict pages that use the shortcodes. Due to how and where the form has to be loaded, we HAVE to load it everywhere for non-standard pages to work at all.

**Comment by mindctrl** [comment]
Can you not check the page content for the shortcode before adding the modal?

**Comment by ghost1227** [comment]
We'd have to run a filter on the page content on every page (not ideal) and it wouldn't be 100% guaranteed. First of all, there's a bunch of shortcodes that COULD be adding download content... EDD core has a few (downloads and purchase_link), FD has its own, plus that wouldn't account for custom themes or templates where they're using the function form.

**Comment by spencerfinnell** [comment]
Instead of enqueuing your scripts (which loads them immediately) you could register them and then only enqueue them here: https://github.com/easydigitaldownloads/edd-free-downloads/blob/master/includes/filters.php#L68

https://developer.wordpress.org/reference/functions/wp_register_script/

**Comment by ghost1227** [comment]
@spencerfinnell It's not the scripts that are the issue. It's the injected hidden form that we add to every page

**Comment by mindctrl** [comment]
Without having looked deeply at it, I wonder if you can only hook in `edd_free_downloads_display_inline()` when this block runs?
 
https://github.com/easydigitaldownloads/edd-free-downloads/blob/master/includes/filters.php#L80

**Comment by ghost1227** [comment]
Nope. Tried it that way originally. The separate function is only there because trying to do the whole thing inline in that one function failed (and resulted in it loading something like ten modals on download list pages)

**Comment by mindctrl** [comment]
Ah ok. I thought WP only ran callbacks attached to hooks once unless they have different priorities.

What about doing what I suggested and also adding a `do_action( 'edd_free_downloads_display_inline' );` at the bottom of that function, and then adding a `did_action` check at the top and returning early if it has already done the action?

I think that would only load the modal markup when needed and only output the markup once.

**Comment by dannyb195** [comment]
https://github.com/easydigitaldownloads/edd-free-downloads/pull/206 should address this issues. While it will still load an empty div on every page it should result in a lighter solution while also removing the jBox.js library. @cklosowski I think we can close this or make it dependent on #152

**Comment by cklosowski** [comment]
Let's leave it open, and when we release #152, we can consider this closed. Since #152 will make it so just the empty wrapper loads, that should take care of the `form` elements from being shown.
